templates/patroni.yml.j2

```diff
@@ -21,9 +21,11 @@ log:
 restapi:
   listen: '{{ self_ip }}:8008'
   connect_address: '{{ self_ip }}:8008'
+  {%- if patroni_password %}
   authentication:
     username: patroni
     password: {{ patroni_password }}
+  {%- endif %}
```

Comment:
If the leader elected executes after the template is rendered we'll be stuck with `None` pass and will be unable to reload via the REST API

Reply:
Can this happen in wild? What is our answer there?

Reply:
I don't think so, but when we add rotation ([DPE-5270](https://warthogs.atlassian.net/browse/DPE-5270)) we should switch to SIGHUPing the service instead of reloading via REST.

Reply:
Basic question: At which point during charm execution the patroni template is rendered?

Reply:
2 @lucasgameiroborges good to have this answer in https://charmhub.io/postgresql-k8s/docs/e-flowchart-charm for VM charm :-D

Reply:
> Basic question: At which point during charm execution the patroni template is rendered?

I don't think there's a set order, especially during charm's bootstrap.

Reply:
I might ask @marceloneppel once hes back, might be good to have this clear and documented somewhere (maybe on the diagrams!)

Reply:
I believe that the issue might be happening at https://github.com/canonical/postgresql-operator/blob/5f546f1251b21b8d7a6b04f03a8bfb71a98d383b/src/upgrade.py#L149, when a non-leader unit is the first one to be updated.

The changes from this PR LGTM.